### PR TITLE
More refactoring

### PR DESCRIPTION
**Pull request by dellaert**
- added eliminate as method of HGFG to test in isolation
- moved HBN test to right place
- added three "bougie" constructors that considerably simplify tests and later examples:
```c++
/// Construct from mean `mu_i` and `sigma_i`.
  HybridGaussianConditional(
      const DiscreteKey mode, Key key,  //
      const std::vector<std::pair<Vector, double>> &parameters);

  /// Construct from conditional mean `A1 p1 + b_i` and `sigma_i`.
  HybridGaussianConditional(
      const DiscreteKey mode, Key key,  //
      const Matrix &A, Key parent,
      const std::vector<std::pair<Vector, double>> &parameters);

  /// Construct from conditional mean `A1 p1 + A2 p2 + b_i` and `sigma_i`.
  HybridGaussianConditional(
      const DiscreteKey mode, Key key,  //
      const Matrix &A1, Key parent1, const Matrix &A2, Key parent2,
      const std::vector<std::pair<Vector, double>> &parameters);
 ```

### CODE REVIEW

File: gtsam/hybrid/HybridGaussianConditional.h
Comment (varunagrawal) [inline gtsam/hybrid/HybridGaussianConditional.h]: I kind of like `discreteParent` more. It is self-documenting.

File: gtsam/hybrid/HybridGaussianConditional.cpp
Comment (varunagrawal) [inline gtsam/hybrid/HybridGaussianConditional.cpp]: Any reason we're doing `const auto&` and not `auto&&`. The latter is usually preferred for argument forward since it accepts all types of constness.

File: gtsam/hybrid/HybridGaussianConditional.cpp
Comment (varunagrawal) [inline gtsam/hybrid/HybridGaussianConditional.cpp:72]: I am guessing you didn't merge the latest develop with my updates?

File: gtsam/hybrid/HybridGaussianFactorGraph.cpp
Comment (varunagrawal) [inline gtsam/hybrid/HybridGaussianFactorGraph.cpp:519]: Would it be better to call `factors.hybridEliminate`? This way we can also do `factors.discreteEliminate` and `factors.continuousEliminate`.

File: gtsam/hybrid/HybridGaussianFactorGraph.cpp
Comment (varunagrawal) [inline gtsam/hybrid/HybridGaussianFactorGraph.cpp:439]: This deviates from the naming scheme of `GaussianFactorGraph` and other graphs, where it is `frontalKeys`. Should we strive for consistency?

File: gtsam/hybrid/tests/testGaussianMixture.cpp
Comment (varunagrawal) [inline gtsam/hybrid/tests/testGaussianMixture.cpp]: We should add a comment about what's going on here, since before it was pretty obvious we were defining a `GaussianMixtureModel`.

File: gtsam/hybrid/tests/testGaussianMixture.cpp
Comment (varunagrawal) [inline gtsam/hybrid/tests/testGaussianMixture.cpp]: Comment about `GaussianMixtureModel`. The function was self-documenting in a way.

**Review by varunagrawal — APPROVED** [review]
Approve modulo comments.

I like the new API but it would also be nice to have at least one test using the old API as a tutorial, aka "this is good, this new thing is better".

File: gtsam/hybrid/HybridGaussianConditional.h
Comment (dellaert) [reply gtsam/hybrid/HybridGaussianConditional.h]: Ok, i’’l fix

File: gtsam/hybrid/HybridGaussianConditional.cpp
Comment (dellaert) [reply gtsam/hybrid/HybridGaussianConditional.cpp]: Sure

File: gtsam/hybrid/HybridGaussianFactorGraph.cpp
Comment (dellaert) [reply gtsam/hybrid/HybridGaussianFactorGraph.cpp:519]: It’s a HFG, eliminate is hybrid by definition.

File: gtsam/hybrid/HybridGaussianFactorGraph.cpp
Comment (dellaert) [reply gtsam/hybrid/HybridGaussianFactorGraph.cpp:439]: Ok, I’ll fix

File: gtsam/hybrid/tests/testGaussianMixture.cpp
Comment (dellaert) [reply gtsam/hybrid/tests/testGaussianMixture.cpp]: Ok

File: gtsam/hybrid/tests/testGaussianMixture.cpp
Comment (dellaert) [reply gtsam/hybrid/tests/testGaussianMixture.cpp]: Ok

**Comment by dellaert** [comment]
I’ll address the comments. But 
- I don’t want the separators between the constructors
- The includes should not depend on transitive inclusion, see Google style guide
Maybe let me edit to avoid conflicts? You have enough to do with experiments :-)

File: gtsam/hybrid/HybridGaussianFactorGraph.cpp
Comment (dellaert) [reply gtsam/hybrid/HybridGaussianFactorGraph.cpp:439]: Actually, no,I went back and looked at GFG and it's always just "keys". So I'll leave this to be consistent. frontal keys is a concept in conditionals, not in elimination. The keys we eliminate will *become* frontal keys.

Comment (varunagrawal) [reply gtsam/hybrid/HybridGaussianFactorGraph.cpp:439]: Oh okay. I must have been mistaken then.

**Comment by varunagrawal** [comment]
Sure thing wrt the dividers. It's a nitpick anyway :)

**Comment by varunagrawal** [comment]
> - The includes should not depend on transitive inclusion, see Google style guide

Interesting. I like to minimize includes since that helps cut down compilation time. Less header copying.

> Maybe let me edit to avoid conflicts? You have enough to do with experiments :-)

Fair. I was updating the code to examine the ordering, so decided to save a compilation cycle by making the edits.

**Pull request merged**
